src/app/treasure-hunt/treasure-hunt-report/treasure-hunt-report.component.ts

```diff
@@ -166,4 +181,45 @@ export class TreasureHuntReportComponent implements OnInit {
     this.printOptionsMenuService.showPrintMenu.next(true);
   }
 
+  showExportModal() {
+    this.fileName = this.getFileName();
+    this.exportModal.show();
+  }
+
+  hideExportModal() {
+    this.exportModal.hide();
+  }
+
+  getFileName(): string {
+    const date: Date = new Date();
+    let formatedDate: string = moment(date).format("MMM D, YYYY").toString();
+    return formatedDate + ' - Treasure Hunt Report.ppt';
+  }
+
+  present() {
+    this.presenting = true;
+    this.cd.detectChanges();
+    if (!this.fileName) {
+      this.fileName = this.getFileName();
+    }
+    let facilityInfo: FacilityInfo;
+    let settings = this.settingsDbService.getByDirectoryId(this.assessment.directoryId);
+    if (settings) {
```

Comment:
This area can be tightened up:
- currently, this logic and createPPT() both check facilityInfo exists. We can just pass settings.facilityInfo into createPPT and do ```if (facilityInfo && facilityInfo.facilityName)```  in that method
- Do we need to check settings exist? Do we need to give feedback if they click to export and there are no settings?